src/libraries/Common/src/System/Security/Cryptography/ECDsaOpenSsl.cs

```diff
@@ -127,67 +147,78 @@ public override bool TrySignHash(ReadOnlySpan<byte> hash, Span<byte> destination
                     return false;
                 }
 
-                ReadOnlySpan<byte> derSignature = SignHash(hash, signDestination, signatureLength, key);
-                bytesWritten = AsymmetricAlgorithmHelpers.ConvertDerToIeee1363(derSignature, KeySize, destination);
-                Debug.Assert(bytesWritten == encodedSize);
+                using (SafeEvpPKeyCtxHandle ctx = CreateCtx())
+                {
+                    ctx.ConfigureForECDSASign();
+
+                    if (!ctx.TryGetSufficientSignatureSizeInBytesCore(hash, out int sufficientSignatureSizeInBytes))
+                    {
+                        throw Interop.Crypto.CreateOpenSslCryptographicException();
+                    }
```

Comment:
This should be captured in Interop, it doesn't need to leak out this far.  (We used to leak it out this far, because we use to only 1:1 shim OpenSSL functions; now we/I have evolved our Interop types to do the bigger work of translating the native concept to the managed concept of an operation).

```suggestion
                    int bufferSizeNeeded = Interop.Crypto.EvpPkeyGetSignatureSize(key);
```

If the problem is that requires an EVP_PKEY_CTX, then we should ask that question just once during key import (same place we set KeySize).

Reply:
I think I forgot we had the [magic formula](https://github.com/dotnet/runtime/blob/main/src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/ECDsa.cs#L1248-L1275) for calculating this (and funny enough I was the one who added that in the PR) but it might still be faster to make `_sign` method calculate it. I'll address this separately after fixing build errors